R/assignment_linter.R

```diff
@@ -2,23 +2,36 @@
 #'
 #' Check that `<-` is always used for assignment.
 #'
+#' @param block_double_assign Logical, `FALSE` by default. If `TRUE`, usage
+#'   of [`<<-`][base::assignOps] and [`->>`][base::assignOps] are also blocked.
 #' @evalRd rd_tags("assignment_linter")
 #' @seealso [linters] for a complete list of linters available in lintr.
 #' @export
-assignment_linter <- function() {
+assignment_linter <- function(block_double_assign = FALSE) {
```

Comment:
IIRC we usually have `allow_*` flags. Also, I'd call this `allow_parent_assign` since double assign could be

```
a <- 1
a <- 2
```

(which is also pointless but harder to prove pointless)

Reply:
used `allow_cascading_assign`